Initial commit

jgleissner opened the pull request:
This is based on the original suma-storage script with a version for SUMa Proxy added.

jgleissner committed:
Initial commit

Based on untracked suma-storage script shipped with SUSE Manager Server
images.
Added script to move squid proxy on SUSE Manager Proxy.

jgleissner committed:
License file is LICENSE not COPYING

jgleissner commented:
@rjschwei Is the implementation fine with you?

rjschwei commented on susemanager-cloud-setup-functions.sh:
```diff
@@ -0,0 +1,134 @@
+create_partition() {
+    test -z "$1" && die "create_parition called without argument"
+    local disk=$1
+    local result=$(yes | parted $disk mklabel msdos 2>&1)
```
We should use gpt to support > 2 TB disks

rjschwei commented on susemanager-cloud-setup-functions.sh:
```diff
@@ -0,0 +1,134 @@
+        die "$part already added to fstab"
+    fi
+    local fs=$(blkid -s TYPE $part -o value)
+    echo "$part $mount_point $fs defaults 1 1" >> /etc/fstab
```
We want "nofail" option in case the VM is rebooted without the device attached. While that's not good it is better than the VM not booting because it cannot find a required file system

Also lets mount by UUID

jgleissner replied:
I only copied the code over from the old script, but that function is actually not used, as it wasn't by the old script either. Instead the documentation instructs the user to add an entry to /etc/fstab, if the framework recommends to add an entry for external storage devices.

Should we change that and automatically add the entry?

jgleissner replied:
@rjschwei I've update the function, but as said, it's not actually used by the setup scripts at the minute. Should I change that or is there a reason why automatically adding the entry might be bad?

rjschwei replied:
@jgleissner now that nofail is fixed and works as intended we can add the fstab entries automatically.

rjschwei commented on susemanager-cloud-setup.spec:
```diff
@@ -0,0 +1,72 @@
+#
+
+
+Name:           susemanager-cloud-setup
```
s/susemanager-cloud-setup/susemanager-cloud-storage-setup/

We do not want to give the impression that this sets up SUMa in any way shape or form

jgleissner replied:
Ok. Should I also rename the repository?

jgleissner replied:
Package name changed. I guess the repository name isn't really that important but I can rename that too.

rjschwei replied:
Yes please change the repo name as well. It's important for those that fish through SUSE-Enceladus

rjschwei commented on susemanager-cloud-setup.spec:
```diff
@@ -0,0 +1,72 @@
+Source:         susemanager-cloud-setup-%{version}.tar.gz
+
+%description
+A simple script that sets up external storage for SUSE Manager.
```
s/simple//

No color commentary ;)

rjschwei commented:
Sorry for the delay in review

jgleissner committed:
use GPT disk label

Use GPT instead of MS-DOS disk label to support volumes >2TB.
Use parted to create partition, as it's more straight-forward than
fdisk.

jgleissner committed:
use UUID for mount it update_fstab

Updated function update_fstab to create fstab entry using UUID.

jgleissner committed:
Rename package to susemanager-cloud-storage-setup

jgleissner committed:
Automatically update /etc/fstab

rjschwei commented on susemanager-cloud-setup-functions.sh:
```diff
@@ -94,11 +94,15 @@ update_fstab() {
     fi
     local fs=$(blkid -s TYPE $part -o value)
-    echo "$part $mount_point $fs defaults 1 1" >> /etc/fstab
+    echo "UUID=$uuid $mount_point $fs defaults 1 2" >> /etc/fstab
```
Still missing "nofail"

jgleissner replied:
I could have sworn I added it. Thanks for catching!

jgleissner committed:
set 'nofail' flag in /etc/fstab entry
